e2e/favorites.spec.ts

```diff
@@ -1,13 +1,120 @@
-import { expect, test } from '@playwright/test'
+import { Page, expect, test } from '@playwright/test'
 import { testAuth } from './fixtures/authenticatedUserPages'
+import { entityUrlPathname } from './helpers/entities'
 import { goToDashboard } from './helpers/testUser'
+import { Project } from './helpers/types'
+import { waitForInitialPageLoad } from './helpers/utils'
+
+const PUBLIC_PROJECT: Project = {
+  name: 'swc-e2e-test-project',
+  id: 'syn12366772',
+}
+
+async function expectFavoritesPageLoaded(page: Page) {
+  await expect(
+    page.getByRole('heading', { name: 'Your Favorites' }),
+  ).toBeVisible()
+  await expect(page.getByPlaceholder('Favorite Name')).toBeVisible()
+}
+
+async function goToFavorites(page: Page) {
+  await page.getByLabel('Favorites', { exact: true }).click()
+  await expectFavoritesPageLoaded(page)
+}
+
+async function expectProjectPageLoaded(
+  page: Page,
+  projectName: string,
+  projectId: string,
+) {
+  await page.waitForURL(entityUrlPathname(projectId))
+  await expect(page.getByRole('heading', { name: projectName })).toBeVisible()
+  await page.waitForURL(
+    `${entityUrlPathname(PUBLIC_PROJECT.id)}/discussion/default`,
+  )
+  await expect(page.getByRole('heading', { name: 'Discussion' })).toBeVisible()
```

Comment:
Wait for navigations to finish and project page to load -- after navigating to the base project page, the user is redirected to `/wiki/` and then `/discussion/default`.